lib/me.robot

```diff
@@ -0,0 +1,30 @@
+*** Comments ***
+@@ -0,0 +1,23 @@
+
+
+*** Settings ***
+Library     BuiltIn
+Library     Process
+
+
+*** Keywords ***
+Check ME State
+    [Documentation]    Keyword Checks The ME PCI Register Offset 0x40 To Find Current ME State, Returns:
+    ...    -Disabled (Soft) For 3
+    ...    -Disabled (HAP) for 2
+    ...    -Enabled for 0
+    TRY
+        ${out}=    Execute Command In Terminal    setpci -s 16.0 40.L
+        ${char}=    Evaluate    ${out}[3]
+        # Current Operation Mode: bits 16:19 of register HFSTS1 (0x40)
+    EXCEPT    Evaluating expression 'p' failed: NameError: name 'p' is not defined nor importable as module
+        ${char}=    Evaluate    2
```

Comment:
I understand that if this command fails (presumably due to no pci device), we assume it is HAP-disabled?

This will not always be the case, but I have no better idea to determine it easily. We can leave this logic like this, but it would be nice to leave some comments, e.g. that we cannot find ME PCI HECI device, so we assume it's HAP-disabled, but it might be soft disabled as well.

However, we rely here on the `setpci` to fail here. There are more cases why it may fail, not only when device does not exist.

So I would suggest, to modfiy this logic as follows:
- check via lspci if ME device exists<
- if not - return it is disabled
- if it does, use the setpci method to determine the actual state

Reply:
the updated keyword https://github.com/Dasharo/open-source-firmware-validation/pull/403/commits/3c4f4589ede0faa0dee81eacb5937e654ea2a784 and the output decoder https://github.com/Dasharo/open-source-firmware-validation/pull/403/commits/dd0ba0cdf1574aac122695d924e03bdf9453ba59